From 81a62b72b95766df2df3b0876e74fda2ed45f8a4 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <jak@debian.org>
Date: Fri, 30 Oct 2015 13:59:21 +0100
Subject: [PATCH] edsp: Use the selected candidate, not the policy candidate

When printing if a version is a candidate, consider the
candidate set in the depcache, and not the candidate in
the policy.

Closes: #803471
---
 apt-pkg/edsp.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apt-pkg/edsp.cc b/apt-pkg/edsp.cc
index 90f2047..8d7a5ae 100644
--- a/apt-pkg/edsp.cc
+++ b/apt-pkg/edsp.cc
@@ -84,7 +84,7 @@ static void WriteScenarioVersion(pkgDepCache &Cache, FILE* output, pkgCache::Pkg
 	   fprintf(output, " %s\n", R->c_str());
    }
    fprintf(output, "APT-Pin: %d\n", Pin);
-   if (Cache.GetCandidateVer(Pkg) == Ver)
+   if (Cache[Pkg].CandidateVerIter(Cache) == Ver)
       fprintf(output, "APT-Candidate: yes\n");
    if ((Cache[Pkg].Flags & pkgCache::Flag::Auto) == pkgCache::Flag::Auto)
       fprintf(output, "APT-Automatic: yes\n");
-- 
2.6.2

